### DISCUSSION TITLE
Serialising Lazy-References

### DISCUSSION THREAD

**Discussion by Bios-Marcel** [eclipse-serializer/serializer#134] [General]
Currently there's a hardcoded check that throws an exception if any `Lazy` element is discovered while serializing an object.

I am rather unsure as to why that is so. Couldn't the serializer simply initialise the lazy reference and then serialize it?

If there's a technical reason as to why this impossible, I think it'd be nice to document this in a discoverable place. Otherwise it might be nice to delete this guard, or explicitly allow activating this via a setter or so, if there are any downsides worth considering for the user of the serializer.

**Comment by hg-ms** [comment]
Hi,
The exception when serializing a lazy reference is intended because lazy references are logically tied to one specific storage if used with Eclipse-Store after they have been persisted the first time. This can cause unpleasant side effects if such a lazy reference is (de)serialized and there is an active Eclipse-Store. That’s why we decided to throw that exception.
Nevertheless, it would be good to be able to serialize such lazy references.

**Reply by Bios-Marcel** [reply]
I am a little confused here. An active eclipse store instance shouldn't be bothered by a separate serialiser instance that deserialises any data, right?

Or would the lazy references automatically get accessed by the storage, due to the fact that the LazyReferenceManager is static?
